source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -236,6 +236,15 @@ TransformerInstance::TransformerInstance(ThreadLocal::Slot &tls, Envoy::Random::
   env_.add_callback("clusterMetadata", 1, [this](Arguments &args) {
     return cluster_metadata_callback(args);
   });
+  env_.add_callback("cluster_metadata", 1, [this](Arguments &args) {
+    return cluster_metadata_callback(args);
+  });
+  env_.add_callback("dynamic_metadata", 1, [this](Arguments &args) {
+    return dynamic_metadata_callback(args);
+  });
+  env_.add_callback("host_metadata", 1, [this](Arguments &args) {
+    return host_metadata_callback(args);
+  });
```

Comment:
I'm not so sure what you mean

Reply:
instead of "dynamic_metadata" as a callback
it might make sense to be: "metadata"

Or do something where we can hook into upstream's command operator framework

Reply:
Can you send me some examples? Didn't you say upstream's metadata impl is very un performant?

Reply:
Oh the current approach is especially as it gets large. Thats actually a good point not to do it. Closing but for reference the header manipulation delegates to substitution formatter which is how it gets its resolution https://github.com/envoyproxy/envoy/blob/release/v1.30/source/common/router/header_parser.cc#L48